src/Stripe.net/Infrastructure/Public/SystemNetHttpClient.cs

```diff
@@ -122,5 +167,62 @@ private static string BuildStripeClientUserAgentString()
 
             return JsonConvert.SerializeObject(values, Formatting.None);
         }
+
+        private static bool ShouldRetry(int numRetries, bool timeout, HttpStatusCode? statusCode)
+        {
+            if (numRetries >= StripeConfiguration.MaxNetworkRetries)
+            {
+                return false;
+            }
+
+            if (timeout == true)
+            {
+                return true;
+            }
+
+            if (statusCode == HttpStatusCode.Conflict)
```

Comment:
It might be worth adding 503s here and non-`POST` 500s as recently came into stripe-node. I think that's the direction we're going, and we just haven't changed stripe-go yet.